### PR TITLE
SIDM-4413 eJudiciary login button and redis session fixes

### PR DESCRIPTION
**Pull request by nikola-naydenov-hmcts**
Adding the eJudiciary login button, fixing redis session and some session quality of life improvement

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java [hunk 1]
```diff
@@ -64,49 +65,7 @@
...
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.STATE;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.TACTICAL_ACTIVATE_VIEW;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.TACTICAL_RESET_PWD_VIEW;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.TERMS_AND_CONDITIONS_VIEW;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.UPLIFT_LOGIN_VIEW;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.UPLIFT_REGISTER_VIEW;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.USERCREATED_VIEW;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.USERNAME;
-import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.VERIFICATION_VIEW;
+import static uk.gov.hmcts.reform.idam.web.helper.MvcKeys.*;
```
Comment (dfourn) [inline src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java:68]: You might want to change your IDE settings

Comment (nikola-naydenov-hmcts) [reply src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java:68]: No

File: src/main/java/uk/gov/hmcts/reform/idam/web/config/SessionConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @Value("${authentication.secureCookie}")
+    private Boolean useSecureCookie;
+
+    @Bean
+    public CookieSerializer cookieSerializer() {
+        DefaultCookieSerializer serializer = new DefaultCookieSerializer();
+        serializer.setCookieName("Idam.SSOSession");
+        serializer.setCookiePath("/");
+        serializer.setDomainNamePattern("^.+?\\.(\\w+\\.[a-z]+)$");
+        serializer.setCookieMaxAge(432000);
```
Comment (radoslaw-orlowski-amido) [inline src/main/java/uk/gov/hmcts/reform/idam/web/config/SessionConfiguration.java:26]: A magic number, can this be moved to a constant for readability?

Comment (nikola-naydenov-hmcts) [reply src/main/java/uk/gov/hmcts/reform/idam/web/config/SessionConfiguration.java]: yeah ill make it a constant

**Pull request merged**